tests/functional/rate_test.go

```
@@ -0,0 +1,47 @@
+package functional
+
+import (
+	"btc-app/config"
+	"btc-app/handler"
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"log"
+	"math"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+)
+
+func TestGetRate(t *testing.T) {
+	rateHandler := handler.NewRateHandler(createConfig())
+	server := httptest.NewServer(rateHandler.GetCurrentRateHandler())
```

To have more control on server responses use mock server instead of real

---

It's discussable question, because I think that E2E test without mocking external API could help us identify when something is going wrong. If consider case when API starts return response in another pattern, we will have to change the logic of handling the new type of response. You are offering mock response, make it hardcoded; and if API change response, our test will pass and it's could lead to error detection on the production instead of during real developing.